core/src/main/java/org/apache/spark/util/collection/unsafe/sort/UnsafeSorterSpillReader.java

```diff
@@ -73,7 +76,9 @@ public void loadNext() throws IOException {
     numRecordsRemaining--;
     if (numRecordsRemaining == 0) {
       in.close();
-      file.delete();
+      if (!file.delete()) {
+        logger.error("Unable to delete spill file {}", file.getPath());
```

Comment:
Is the file going to be cleaned later ?

Do you have any suggestion w.r.t. what log level should be used ?